service/provider/src/main/java/gov/va/vro/service/provider/mas/service/MasCollectionService.java

```diff
@@ -93,55 +89,39 @@ public static HealthDataAssessment combineEvidence(
     AbdEvidence masApiEvidence = masApiAssessment.getEvidence();
     // for now, we just add up the lists
-    log.info("combineEvidence >> LH  : " + lighthouseEvidence != null ? "not null" : "null");
-    log.info("combineEvidence >> MAS : " + masApiEvidence != null ? "not null" : "null");
+    log.info("combineEvidence >> LH  : " + ((lighthouseEvidence != null) ? "not null" : "null"));
+    log.info("combineEvidence >> MAS : " + ((masApiEvidence != null) ? "not null" : "null"));
     AbdEvidence compositeEvidence = new AbdEvidence();
     compositeEvidence.setBloodPressures(
         merge(
-                lighthouseEvidence != null ? lighthouseEvidence.getBloodPressures() : null,
-                masApiEvidence != null ? masApiEvidence.getBloodPressures() : null)
-            .stream()
-            .flatMap(s -> Stream.ofNullable(s))
-            .distinct()
-            .collect(Collectors.toList()));
+            lighthouseEvidence != null ? lighthouseEvidence.getBloodPressures() : null,
+            masApiEvidence != null ? masApiEvidence.getBloodPressures() : null));
     compositeEvidence.setConditions(
         merge(
-                lighthouseEvidence != null ? lighthouseEvidence.getConditions() : null,
-                masApiEvidence != null ? masApiEvidence.getConditions() : null)
-            .stream()
-            .flatMap(s -> Stream.ofNullable(s))
-            .distinct()
-            .collect(Collectors.toList()));
+            lighthouseEvidence != null ? lighthouseEvidence.getConditions() : null,
+            masApiEvidence != null ? masApiEvidence.getConditions() : null));
     compositeEvidence.setMedications(
         merge(
-                lighthouseEvidence != null ? lighthouseEvidence.getMedications() : null,
-                masApiEvidence != null ? masApiEvidence.getMedications() : null)
-            .stream()
-            .flatMap(s -> Stream.ofNullable(s))
-            .distinct()
-            .collect(Collectors.toList()));
+            lighthouseEvidence != null ? lighthouseEvidence.getMedications() : null,
+            masApiEvidence != null ? masApiEvidence.getMedications() : null));
     compositeEvidence.setProcedures(
         merge(
-                lighthouseEvidence != null ? lighthouseEvidence.getProcedures() : null,
-                masApiEvidence != null ? masApiEvidence.getProcedures() : null)
-            .stream()
-            .flatMap(s -> Stream.ofNullable(s))
-            .distinct()
-            .collect(Collectors.toList()));
+            lighthouseEvidence != null ? lighthouseEvidence.getProcedures() : null,
+            masApiEvidence != null ? masApiEvidence.getProcedures() : null));
     lighthouseAssessment.setEvidence(compositeEvidence);
 
     return lighthouseAssessment;
   }
 
   private static <T> List<T> merge(List<T> list1, List<T> list2) {
-    List<T> result = new ArrayList<>();
+    Set<T> result = new LinkedHashSet<>();
```

Comment:
It is good to use Set to avoid dup. Will it be better to have Lighthouse and MAS services return SETs and avoid to have this "merge" process? On the other hand, those Abd Evidence objects have a comparable defined. I'm not sure whether this will have an unintended impact.

Reply:
Even if they return sets, we'd have to combine them, so there is a still a merge. Set does not care about comparable, only equals and hashCode.

Ideally, comparable equality should be compatible with equals() equality. I don't think this is the case here, so perhaps instead of making them Comparable we should use a Comparator. That's a different topic.